contribute/architecture/k8s-inspired-backend-arch.md

```diff
@@ -0,0 +1,288 @@
+# Grafana Kubernetes-inspired backend architecture
+
+Grafana is adopting a Kubernetes-inspired, resource-oriented model for all Grafana core resources (dashboards, folders, datasources, alert rules, ..) where users can interact with these resources by defining their desired state via well-structured objects sent to Grafana's new Resource APIs (`/apis/...`). Grafana then acts upon these declarations to align the system's current state with the requested state.
+
+On top of adopting the declarative, eventually-consistent system design, we are also adopting Kubernetes API's strict approach to URL structures, versioning and schemas. This approach addresses limitations of Grafana's traditional Legacy APIs (`/api/...`), including:
+- **Inconsistent designs:** Variations in structure, parameters, and response formats across different endpoints.
+- **No versioning:** Difficulty managing breaking changes and API evolution without clear versioning.
+- **Absence of schemas:** Lack of machine-readable schemas hindered programmatic interaction, validation, and tooling.
+- **Limited extensibility model:** No standardized way for Grafana plugins to extend core APIs.
+
+The end goal is for the Resource APIs to become the only interface for managing core Grafana resources. We will eventually deprecate and remove all Legacy API endpoints.
+
+## 1. Key Differences from Legacy API Endpoints
+
+- **URL structure & versioning:**
+  - **Resource APIs:** Follow Kubernetes conventions (`/apis/<group>/<version>/namespaces/<namespace>/<resource>/<name>`). Includes explicit API versions (e.g., `v0alpha1`, `v1alpha1`, `v2alpha1`) in the path, allowing for controlled evolution and multiple versions of a resource API to co-exist.
```

Comment:
Maybe give examples of beta & stable too (e.g., `v0alpha1`, `v1`, `v2beta1`)?

Reply:
done